src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -93,12 +93,13 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_copy_ep_get_zcopy,
 
     UCT_CUDA_COPY_CHECK_AND_CREATE_STREAM(iface->stream_d2h);
 
-    VALGRIND_MAKE_MEM_DEFINED(iov[0].buffer, iov[0].length);
-
     status = uct_cuda_copy_post_cuda_async_copy(tl_ep, iov[0].buffer, (void *)remote_addr,
                                                 iov[0].length, cudaMemcpyDeviceToHost,
                                                 iface->stream_d2h,
                                                 &iface->outstanding_d2h_cuda_event_q, comp);
+    if (status == UCS_INPROGRESS) {
```

Comment:
maybe better `if (!UCS_STATUS_IS_ERR(status))?` (to not depend on `uct_cuda_copy_post_cuda_async_copy` behavior)